bin/record-build-info.ts

```diff
@@ -5,8 +5,9 @@ import { resolve } from 'path'
 const target = resolve(process.cwd(), 'build-info.json')
 
 function getEnvironmentVariable(name: string): string {
-  if (process.env[name]) {
-    return process.env[name]
+  const value = process.env[name]
+  if (value) {
+    return value
```

Comment:
I'm probably missing something, aren't the two equivalent?

is TS not able to infer the `string` type if `process.env[name]` is not assigned to a variable?

Reply:
(I'm asking mainly for my benefit)

Reply:
they are identical to me too! but in strict mode, typescript didn't "narrow" the type of `process.env[name]` with the if statement. maybe because it's feasible for the value to change in the meantime? possibly typescript isn't being clever enough to notice

Reply:
Oh cool, so it seems like TS needs a variable to correctly infer the type.

Actually my guess would be that keep track of all the types of all the properties of all objects may potentially be overwhelming? It doesn't help that this is an expression and kinda dynamic as it's a method call (I think...)

Yeah, it seems like in this case TS is not clever enough to figure it out.

Reply:
i did a vaguely-similar thing here and narrowing worked 🤷 

https://github.com/ministryofjustice/manage-my-prison/pull/43/files#diff-3e741135921d46fa876e82f2d84566bf7e0693628fe9107c28ad2dd97084c22dR33